osdep/timer-win32.c

```diff
@@ -27,28 +29,35 @@
 
 static LARGE_INTEGER perf_freq;
 
-// ms values
-static int hires_max = 50;
-static int hires_res = 1;
+// ns values
+static int64_t hires_max = MP_TIME_US_TO_NS(50000);
+static int64_t hires_res = MP_TIME_US_TO_NS(500);
 
-int mp_start_hires_timers(int wait_ms)
+// NtSetTimerResolution allows setting the timer resolution to less than 1 ms.
+// Resolutions are specified in 100-ns units.
+// If Set is TRUE, set the RequestedResolution. Otherwise, return to the previous resolution.
+NTSTATUS NtSetTimerResolution(ULONG RequestedResolution, BOOLEAN Set, PULONG ActualResolution);
+
+int64_t mp_start_hires_timers(int64_t wait_ns)
 {
 #if !HAVE_UWP
+    ULONG actual_res = 0;
     // policy: request hires_res ms resolution if wait < hires_max ms
-    if (wait_ms > 0 && wait_ms <= hires_max &&
-        timeBeginPeriod(hires_res) == TIMERR_NOERROR)
+    if (wait_ns > 0 && wait_ns <= hires_max &&
+        NtSetTimerResolution(hires_res / 100, TRUE, &actual_res) == STATUS_SUCCESS)
```

Comment:
You actually need to check `NtQueryTimerResolution` for the supported values. `timeBeginPeriod` does that for use clamping to minimum. Else we will end up with failing `NtQueryTimerResolution` instead of setting min supported period.